jhotdraw-core/src/main/java/org/jhotdraw/draw/figure/BezierFigure.java

```diff
@@ -184,24 +185,23 @@ public boolean contains(Point2D.Double p, double scaleDenominator) {
               AttributeKeys.getStrokeTotalWidth(this, scaleDenominator)
                   * attr().get(STROKE_MITER_LIMIT));
       if (gs.createStrokedShape(path).contains(p)) {
-        return true;
+        result = true;
       } else {
         if (isClosed()) {
           return false;
         }
       }
-    }
-    if (!isClosed()) {
+    } else if (!isClosed()) {
       if (getCappedPath(scaleDenominator).outlineContains(p, tolerance)) {
-        return true;
+        result = true;
       }
       if (attr().get(START_DECORATION) != null) {
```

Comment:
The same problem with the return statements can be found here. Lines 198 to 215 in the original file should be in an else statement.